components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/test/java/org/wso2/carbon/identity/api/resource/mgt/dao/APIResourceManagementDAOImplTest.java

```diff
@@ -139,16 +163,33 @@ public void testAddAPIResource(String postfix, int tenantId) throws Exception {
     public Object[][] getScopesByAPIData() {
         // Define your test cases here
         return new Object[][]{
-                {"GetScopesTest", TENANT_ID, 2},
-                {"GetScopesTest2", INVALID_TENANT_ID, 0}
+                {"GetScopesTest", TENANT_ID, false, 2},
+                {"GetScopesTest2", INVALID_TENANT_ID, false, 0},
+                {"GetScopesTest3", TENANT_ID, true, 2}
         };
     }
 
     @Test(dataProvider = "getScopesByAPIData", priority = 3)
-    public void testGetScopesByAPI(String name, Integer tenantId, int expected) throws Exception {
+    public void testGetScopesByAPI(String name, Integer tenantId, boolean isOrganization, int expected)
+            throws Exception {
 
-        try (MockedStatic<IdentityDatabaseUtil> identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class)) {
-            String apiId = addAPIResourceToDB(name, getConnection(), tenantId, identityDatabaseUtil).getId();
+        try (MockedStatic<IdentityDatabaseUtil> identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class);
+             MockedStatic<OrganizationManagementUtil> organizationManagementUtil =
+                     mockStatic(OrganizationManagementUtil.class);
+             MockedStatic<IdentityTenantUtil> identityTenantUtil = mockStatic(IdentityTenantUtil.class)) {
+            if (isOrganization) {
+                identityTenantUtil.when(() -> IdentityTenantUtil.getTenantDomain(anyInt())).thenReturn("tenant1");
+                OrganizationManager organizationManager = mock(OrganizationManager.class);
+                APIResourceManagementServiceComponentHolder.getInstance().setOrganizationManager(organizationManager);
+                lenient().when(organizationManager.resolveOrganizationId(anyString())).thenReturn("org-id-1234");
+                lenient().when(organizationManager.getPrimaryOrganizationId("org-id-1234"))
+                        .thenReturn("prim-org-id-1234");
+                lenient().when(organizationManager.resolveTenantDomain("prim-org-id-1234"))
+                        .thenReturn("prim-tenant");
+                identityTenantUtil.when(() -> IdentityTenantUtil.getTenantId(anyString())).thenReturn(tenantId);
+            }
+            String apiId = addAPIResourceToDB(name, getConnection(), tenantId, identityDatabaseUtil,
+                    organizationManagementUtil, isOrganization).getId();
```

Comment:
we are not allowing to add API resources for sub orgs right?
As per this test, you are adding the API resources to the sub org tenant id